Impl proposal page

malloc3141 opened the pull request:
# 요약 \*

<!-- 닫는 이슈 번호 및 PR 내용에 대한 간단한 요약 표기. -->

It closes #73 

# 스크린샷

<!-- PR 변경 사항에 대한 스크린샷이나 .gif 파일 -->

아직 수정중입니다! #72 에서 구현되고 있는 single column table 머지되면 조회 가이드 부분 만들듯 합니다

- [x] 만들어 둔 테이블에 태그나 아이콘버튼 넣기
- [x] 근거 버튼 컴포넌트 만들기
- [ ] 조회 가이드 컴포넌트 구현

# 이후 Task \*

<!-- PR 이후 개설할 이슈 목록 -->

- 없음

malloc3141 committed:
feat: impl skeleton proposal page

malloc3141 committed:
feat: impl tables

malloc3141 commented:
- [ ] 백엔드 단에서 enum 불러와서 쓰고 있는데 승인 상태 관련 enum이랑 수입/지출/총계 enum이 없어서 만들어지면 적용
- [ ] 뷰어 권한이랑 수정 권한에서 근거랑 비고 모달이 달라야되는데 아직은 뷰어만 되어있음
- [ ] 지출 테이블에서 호버 시 모달 뜨는거 지금은 mui tooltip으로 해놨는데 디자인 맞춰서 수정
- [ ] 조회 가이드 single column table 머지되면 그거 이용해서 구현
- [ ] ViewResult(맨 위 표) 디자인 맞게 수정

malloc3141 commented:
hydration failed because the initial ui does not match what was rendered on the server 에러가 떠서(테이블 컴포넌트에서 에러가 나는듯한데...) 그걸 해결하려다가 시간을 많이 날려먹었는데 해결을 못했습니다 ㅜㅜ
일단 리뷰 요청을 하고 저녁에 다시 고쳐볼게용...

malloc3141 committed:
fix: fix build error

wjeongchoi commented on packages/web/src/app/proposal/page.tsx:
```diff
@@ -0,0 +1,63 @@
+  // const [organization, setOrganization] = useState<string>("");
+
+  <FlexWrapper direction="column" gap={16}>
+    <Typography
+      fs={30}
+      lh={40}
+      color="GREEN.800"
+      fw="SEMIBOLD"
+      style={{ padding: "0 80px" }}
+    >
+      예결산 조회
+    </Typography>
```
이 부분 클럽스 PageHead 가지고 와서 쓰면 좋을 것 같아요

wjeongchoi commented on packages/web/src/features/proposal/components/ExpenditureTable.tsx:
```diff
@@ -0,0 +1,209 @@
+import { useFormatter } from "next-intl";
+import DetailButton from "@sparcs-students/web/features/proposal/components/_atomic/DetailButton";
+import DarkTag from "@sparcs-students/web/common/components/Tag/DarkTag";
+import { Tooltip } from "@mui/material";
```
기본 툴팁 말고 따로 구현해서 써줄 수 있나요

wjeongchoi commented on packages/web/src/common/components/Tag/Tag.tsx:
```diff
@@ -0,0 +1,107 @@
+  width?: string;
+}
+
+const Tag: React.FC<TagProps> = ({
```
DarkTag랑 구분해서 쓰려면 LightTag로 쓰면 좋을 것 같아요

wjeongchoi commented on packages/web/src/features/proposal/components/ViewResult.tsx:
```diff
@@ -0,0 +1,92 @@
+];
+
+const ViewResult: React.FC<ViewResultProps> = ({
+  fileName = "전산학부 24년도 예산안",
+  organization = "전산학부",
+  period = "2024년도 하반기",
+  headPerson = "김스튜",
+  submitDate = new Date(),
```
이 부분 mock data 쓰려면 컴포넌트 자체에 쓰지 말고, 외부에서 넣어주세요

wjeongchoi reviewed (commented):
디자인 디테일한 수정은 좀 더 들어가야 할 것 같아요

malloc3141 committed:
feat: refactor Tag component to LightTag, impl custom tooltip

malloc3141 commented:
수정했어요! 혹시 요 에러는 어떻게 고치는지 아시나요..??
![image](https://github.com/user-attachments/assets/4d6e203f-a54a-407a-ade5-343efe8b0b25)

wjeongchoi committed:
fix: fix prop error

wjeongchoi committed:
fix: fix overlay error

wjeongchoi commented:
https://github.com/user-attachments/assets/555dc726-2e88-4c0c-ab6e-a70651af82d3

여기 보면 로딩되는 과정에서 ui가 한 번 깨지는데 이 부분 때문에 문제 생길거에요
다른 마이너한 에러 두 개는 에러 보는 과정에서 바로 고쳤어요

wjeongchoi committed:
fix: build error

wjeongchoi committed:
Merge branch 'dev' into 73-impl-proposal-page

malloc3141 committed:
fix: fix hydration error

malloc3141 commented:
[ae50f94](https://github.com/academic-relations/ar-007-students/pull/74/commits/ae50f9451a4fbf20f339b7b6eef159168113d7bb) 여기 있는 컴포넌트들 제가 고친 적이 없는데 갑자기 왜 문제가 생겼을까요..?? 고쳐주셨다니 다행이네용
hydration 에러는 고쳤습니다

malloc3141 committed:
chore: merge with dev branch

wjeongchoi committed:
fix: fix pnpm lock

ChaeyeonAhn commented on packages/web/src/common/components/Table.tsx:
테이블 스타일이 디자인이랑 다른 것 같아요.

예를 들면 코드 같은 경우에 대충 요렇게 생겼는데 

display: flex;
width: 80px;
padding: 8px 12px;
justify-content: space-between;
align-items: center;

헤더 패딩이 디자인에 비해 큰 것 같고 border-radius도 8인 것 같은데 현재 디자인은 4거든요

바뀐 디자인이 많아서 차근차근 보시면서 반영해주셔야 할 것 같아요.. 😭 너무 많겠지만 화이팅입니다 ㅠㅠㅠ

ChaeyeonAhn commented on packages/web/src/common/components/Table.tsx:
```diff
@@ -0,0 +1,190 @@
+  min-width: ${({ minWidth }) => `max(100%, ${minWidth}px)`};
+  width: fit-content;
+  border: 1px solid ${({ theme }) => theme.colors.GRAY[200]};
+  border-radius: 8px;
```
이거 4px로 수정하면 될 듯 해요

ChaeyeonAhn commented on packages/web/src/common/components/Table/TableCell.tsx:
CommonCellWrapper의 height: 48px 이거 없애야 해요.
패딩도 헤더의 경우 12 8이 아니고 8 12인가 그럴 거에요 / content의 경우는 12 20이고
그리고 기타 다른 것도 여기랑 비교해서 디자인 수정해주시면 많이 해결되지 않을까 싶습니다!

ChaeyeonAhn replied:
아마 여기 때문에 좀 주요하게 디자인이랑 달라지는 듯

malloc3141 replied:
이거 height: 48px 빼면 
![image](https://github.com/user-attachments/assets/bb81fbe7-12cd-4004-9682-c70801261109)
요렇게 높이가 엄청 늘어져서 보여용

ChaeyeonAhn replied:
혹시 아니면 fit-content나 이런걸로 해봐줄수있어요? 제가지금 밖이라.. 내일낮에 저도 한번 볼게요

ChaeyeonAhn commented on packages/web/src/features/proposal/components/IncomeTable.tsx:
```diff
@@ -0,0 +1,192 @@
+          return <LightTag color="GRAY">-</LightTag>;
+      }
+    },
+    size: 80,
```
이 사이즈 80이 분명 맞거든요? 근데 왜 디자인이랑 달라지는지 모르겠어요 디자인으로는 훨씬 좁은데 왜케 넓지 흐어어

wjeongchoi replied:
cell이 피그마보다 훨씬 좁게 설정해야 비슷하게 나오는 것 같더라고요...
비율만 유지하면서 수치 좀 줄여보고 테스트해보는게 좋을 것 같아요

malloc3141 replied:
근데 이거 수치 조절을 좀 해도 화면 크기에 따라서 수입/지출 테이블은 행 전체가 다 안담길 수도 있을 것 같아용(총합 테이블은 문제 없음)

malloc3141 replied:
옹 밑에 적혀있는대로 패딩을 빼니 해결이 되었슴당 근데 이제는 정렬이 망가져버렸어요..

ChaeyeonAhn replied:
ㅠㅠ네네!!
흐어어.. 제가함 보겠습니다

ChaeyeonAhn commented on packages/web/src/app/proposal/page.tsx:
```diff
@@ -0,0 +1,50 @@
+          <Button buttonText="조회" style={{ marginLeft: "auto" }} />
+        </FlexWrapper>
+      </FlexWrapper>
+      <FlexWrapper direction="column" gap={32} style={{ padding: "0 100px" }}>
```
이건 전적으로 제 잘못인데 이미 페이지에 기본적으로 반응형에 따른 padding 설정이 돼 있어서 여기서 padding: "0 100px"를 빼도 될 거에요. 그러면 테이블이 좀 더 가로로 길어짐서 한 눈에 들어오는 형태가 될 거에요.
그래도 지금 테이블에 가로 스크롤 생기면 안 되고, 코드 부분 열이 너무 넓은 건 해결을 해야 할 것 같습니다..!
```suggestion
      <FlexWrapper direction="column" gap={32}>
```

malloc3141 replied:
![image](https://github.com/user-attachments/assets/7f61247b-c78f-40d0-855e-cfd5f6a48b2a)
그부분 패딩을 빼면 이렇게 되어버려요..

ChaeyeonAhn replied:
오마이.... 알겠습니다ㅠ 이거는 제가 디자인 고친걸 빠른시일내에 드릴게요

ChaeyeonAhn reviewed (commented):
디자인 어디어디 수정하면 좋을지 적어 놓았는데, 이것 외에도 특히 ContentRow 속 셀 하나하나 너비를 한 번 더 봐야 할 것 같아요!

malloc3141 committed:
feat: fix design issues

malloc3141 committed:
merge with remote
Merge branch '73-impl-proposal-page' of https://github.com/academic-relations/ar-007-students into 73-impl-proposal-page

ChaeyeonAhn commented:
어근데 매니저까지 다하는거에요? 아직 매니저는 확정이 아닌걸로 알고있는데

malloc3141 commented:
밑에 리뷰어랑 매니저 둘다 있길래 다 하는건줄 알았어용 이건 @wjeongchoi 님이 봐주셔야 할듯??

malloc3141 committed:
feat: setup modal

wjeongchoi commented on packages/web/src/app/proposal/page.tsx:
```diff
@@ -0,0 +1,50 @@
+  // const [documentType, setDocumentType] = useState<string>(""); // 예산안, 결산안, 사업보고서, 사업계획서
+  // const [organization, setOrganization] = useState<string>("");
+
+  <FlexWrapper direction="column" gap={16}>
```
```suggestion
  <FlexWrapper direction="column" gap={48}>
```

wjeongchoi commented on packages/web/src/app/proposal/page.tsx:
```diff
@@ -0,0 +1,50 @@
+
+  <FlexWrapper direction="column" gap={16}>
+    <PageTitle>예결산 조회</PageTitle>
+    <FlexWrapper direction="column" gap={60} style={{ padding: "32 0px" }}>
```
padding 32 아닌 것 같아요

malloc3141 replied:
![image](https://github.com/user-attachments/assets/b77bca24-fee0-44a1-abfe-72f277590abc)
엇 이거 아닌가요??

wjeongchoi replied:
이거 어느 부분 본건지 좀이따 구두로 알려주세요!

wjeongchoi commented on packages/web/src/app/proposal/page.tsx:
```diff
@@ -0,0 +1,50 @@
+  <FlexWrapper direction="column" gap={16}>
+    <PageTitle>예결산 조회</PageTitle>
+    <FlexWrapper direction="column" gap={60} style={{ padding: "32 0px" }}>
+      <FlexWrapper direction="column" gap={32} style={{ padding: "0 100px" }}>
```
이거 100은 넣은 이유가 혹시 뭘까요

malloc3141 replied:
https://github.com/academic-relations/ar-007-students/pull/74#discussion_r1918089481 빼면 저렇게 돼요

wjeongchoi replied:
앗 확인했습니다

wjeongchoi commented on packages/web/src/features/proposal/components/SearchThreeInput.tsx:
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+import FlexWrapper from "@sparcs-students/web/common/components/FlexWrapper";
+
+const SearchThreeInput = () => <FlexWrapper direction="column" gap={16} />;
+
+export default SearchThreeInput;
```
얘는 헷갈리니까 지웁시다

wjeongchoi commented on packages/web/src/features/proposal/components/TotalTable.tsx:
```diff
@@ -0,0 +1,130 @@
+  columnHelper.accessor("type", {
+    id: "type",
+    header: "분류",
+    cell: info => {
+      switch (info.getValue()) {
+        case "수입":
+          return <LightTag color="GREEN100">{info.getValue()}</LightTag>;
+        case "지출":
+          return <LightTag color="GREEN600">{info.getValue()}</LightTag>;
+        case "총계":
+          return <LightTag color="GREEN800">{info.getValue()}</LightTag>;
+        default:
+          return <LightTag color="GRAY">-</LightTag>;
+      }
+      // TODO: use enum for return
```
이 부분 별도 함수로 만들어서 color / text 가져올 수 있으면 좋을 것 같아요

wjeongchoi commented on packages/web/src/features/proposal/components/TotalTable.tsx:
```diff
@@ -0,0 +1,130 @@
+    id: "ratio",
+    header: "비율",
+    cell: info => {
+      if (info.getValue() > 100) {
+        return (
+          <LightTag color="CHERRY">{info.getValue().toFixed(1)}%</LightTag>
+        );
+      }
+      if (info.getValue() <= 100) {
+        return (
+          <LightTag color="THISTLE">{info.getValue().toFixed(1)}%</LightTag>
+        );
+      }
+      return <LightTag color="GRAY">-</LightTag>;
+    },
```
이 부분도 이곳저곳 쓰일 것 같아서 별도로 뺍시다

wjeongchoi commented on packages/web/src/features/proposal/components/ViewerExpenditureTable.tsx:
```diff
@@ -0,0 +1,208 @@
+    id: "code",
+    header: "코드",
+    cell: info => {
+      switch (Math.trunc(info.getValue() / 100)) {
+        case 1:
+        case 4:
+          return <LightTag color="BLUE">{info.getValue()}</LightTag>;
+        case 2:
+        case 5:
+          return <LightTag color="YELLOW">{info.getValue()}</LightTag>;
+        case 3:
+        case 6:
+          return <LightTag color="PINK">{info.getValue()}</LightTag>;
+        default:
+          return <LightTag color="GRAY">-</LightTag>;
+      }
```
이것도 별도 함수로 뺍시다

wjeongchoi commented on packages/web/src/features/proposal/components/_atomic/ExpenditureHelpButton.tsx:
이거 뜰 때 아래 표 안 밀리게 고치면 좋을 것 같아요

wjeongchoi reviewed (commented):
전반적으로 태그 관련 switch나 if문이 많은데 다른데서도 활용되는 요소니까 feature/documents 아래에 utils 폴더 만들고 그 안에다 관련 함수 넣으면 좋을 것 같아요

malloc3141 committed:
Update packages/web/src/app/proposal/page.tsx

Co-authored-by: Woojeong Choi <144689054+wjeongchoi@users.noreply.github.com>

malloc3141 committed:
feat: remove SearchThreeInput

malloc3141 committed:
feat: remove SearchThreeInput legacy

malloc3141 committed:
merge with remote
Merge branch '73-impl-proposal-page' of https://github.com/academic-relations/ar-007-students into 73-impl-proposal-page

malloc3141 committed:
feat: move tag functions to util

malloc3141 committed:
fix: fix hover modal position

malloc3141 committed:
fix: fix pnpm lock conflict

malloc3141 committed:
fix: fix build error

wjeongchoi reviewed (approved):
굿 좋아요

wjeongchoi committed:
Merge branch 'dev' into 73-impl-proposal-page
